### PR TITLE
Backend feature/admin ban user delete comment

### PR DESCRIPTION
**Pull request by 16omerak**

### CODE REVIEW

File: backend/tursu_admin/views.py [hunk 1]
```diff
@@ -64,3 +65,30 @@ def verify_product(request):
...
+@api_view(['POST'])
+def ban_user(request):
+    """Bans given user."""
+    user = get_vendor_from_request(request)
+    if user == None:
+        user = get_customer_from_request(request)
+        if user == None:
+            return HttpResponse("There is no such user.", status=400)
+    user.user.is_banned = True
+    user.save()
```
Comment (onurkilic2301) [inline backend/tursu_admin/views.py:80]: user.user has a field changed here but user is saved. user.user should be saved instead. Also the variable naming can be changed for cleaner code.

**Review by onurkilic2301 — CHANGES_REQUESTED** [review]

File: backend/tursu_admin/views.py (hunk 1 above)
Comment (16omerak) [reply backend/tursu_admin/views.py]: Necessary corrections are made.

File: backend/tursu_admin/views.py [hunk 2]
```diff
@@ -64,3 +65,30 @@ def verify_product(request):
     product.is_verified = True
     product.save()
     return HttpResponse("Successfully verified product")
+
+@authentication_classes([SessionAuthentication, BasicAuthentication])
+@permission_classes((IsAuthenticated,))
+@api_view(['POST'])
+def ban_user(request):
```
Comment (onurkilic2301) [inline backend/tursu_admin/views.py:72]: 1. Admin authentication should be done.
2. At least one parameter to find the respective customer or vendor should be taken. My suggestion would be to take email or username which are common in both user types and then the retrieval can be done in the function.

File: backend/tursu_admin/views.py [hunk 3]
```diff
@@ -64,3 +65,30 @@ def verify_product(request):
...
+        if user == None:
+            return HttpResponse("There is no such user.", status=400)
+    user.user.is_banned = True
+    user.save()
+    return HttpResponse("success")
+    
+@authentication_classes([SessionAuthentication, BasicAuthentication])
+@permission_classes((IsAuthenticated,))
+@api_view(['DELETE','POST'])
+def delete_comment(request):
```
Comment (onurkilic2301) [inline backend/tursu_admin/views.py:92]: 1. Admin authentication should be done.

**Review by onurkilic2301 — CHANGES_REQUESTED** [review]

File: backend/tursu_admin/views.py [hunk 4]
```diff
@@ -71,20 +71,26 @@ def verify_product(request):
...
-    if bannedUser == None:
-        bannedUser = get_customer_from_request(request)
-        if bannedUser == None:
-            return HttpResponse("There is no such user.", status=400)
-    bannedUser.user.is_banned = True
-    bannedUser.user.save()
+    admin = get_admin_from_request(request)
+    if(admin is None):
+        return HttpResponse("Admin authentication failed", status=401)
+    username = request.POST["username"]
```
Comment (onurkilic2301) [inline backend/tursu_admin/views.py]: The case in which username parameter is not given should be handled.

File: backend/tursu_admin/views.py (hunk 2 above)
Comment (16omerak) [reply backend/tursu_admin/views.py:72]: Necessary changes are made.

File: backend/tursu_admin/views.py (hunk 3 above)
Comment (16omerak) [reply backend/tursu_admin/views.py:92]: Necessary changes are made.

File: backend/tursu_admin/views.py (hunk 4 above)
Comment (16omerak) [reply backend/tursu_admin/views.py]: Necessary changes are made.

**Review by onurkilic2301 — APPROVED** [review]

**Pull request merged**